pkg/server/api/svid/v1/service.go

```diff
@@ -24,15 +35,73 @@ type Service interface {
 }
 
 // New creates a new SVID service
-func New(config Config) Service {
-	return &service{}
+func New(config *Config) Service {
+	return &service{
+		ServerCA:    config.ServerCA,
+		TrustDomain: config.TrustDomain,
+	}
 }
 
 type service struct {
+	ServerCA    ca.ServerCA
+	TrustDomain spiffeid.TrustDomain
 }
 
 func (s *service) MintX509SVID(ctx context.Context, csr *x509.CertificateRequest, ttl time.Duration) (*api.X509SVID, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	log := rpccontext.Logger(ctx)
+
+	if err := csr.CheckSignature(); err != nil {
+		log.WithError(err).Error("Invalid CSR: signature verify failed")
+		return nil, status.Errorf(codes.InvalidArgument, "invalid CSR: signature verify failed")
+	}
+
+	if len(csr.URIs) != 1 {
+		log.Error("Invalid CSR: a valid URI is required")
+		return nil, status.Errorf(codes.InvalidArgument, "invalid CSR: a valid URI is required")
+	}
+
+	spiffeID, err := spiffeid.FromURI(csr.URIs[0])
+	if err != nil {
+		log.WithError(err).Error("Invalid SPIFFE ID")
+		return nil, status.Errorf(codes.InvalidArgument, "invalid SPIFFE ID: %v", err)
+	}
+
+	if !spiffeID.MemberOf(s.TrustDomain) {
+		log.Error("Invalid SPIFFE ID: not member of servers trust domain")
+		return nil, status.Error(codes.InvalidArgument, "invalid SPIFFE ID: not member of servers trust domain")
+	}
+
+	for _, dnsName := range csr.DNSNames {
+		if err := validateDNS(dnsName); err != nil {
+			log.WithError(err).Error("Invalid DNS name")
+			return nil, status.Errorf(codes.InvalidArgument, "invalid DNS name: %v", err)
+		}
+	}
+
+	var subject *pkix.Name
+	if csr.Subject.String() != "" {
+		subject = &csr.Subject
+	}
+
+	svid, err := s.ServerCA.SignX509SVID(ctx, ca.X509SVIDParams{
+		SpiffeID:  spiffeID.String(),
+		PublicKey: csr.PublicKey,
+		// SignX509SVID is taking care of of ttl comparation against bundle
+		TTL:     ttl,
+		DNSList: csr.DNSNames,
+		Subject: subject,
+	})
+	if err != nil {
+		log.WithError(err).Error("Failed to sign X.509 SVID")
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	// TODO: may I verify svid is not empty?
+	return &api.X509SVID{
+		ID:        spiffeID,
+		CertChain: svid,
+		ExpiresAt: svid[0].NotAfter.Local().UTC(),
```

Comment:
Curious why you convert to `Local()` before converting to `UTC()`? The certificate `notAfter` should already be in UTC, but if we want to be defensive, we can just call `UTC()` without first going to `Local()`...
```suggestion
		ExpiresAt: svid[0].NotAfter.UTC(),
```

Reply:
that was a typo.. my idea was always do just a `UTC`